extensions/wikia/Wall/WallExternalController.class.php

```diff
@@ -666,9 +687,18 @@ public function replyToMessage() {
 		/** @var $wallMessage WallMessage */
 		$wallMessage = WallMessage::newFromTitle( $parentTitle );
 		$body = $this->getConvertedContent( $this->request->getVal( 'body' ) );
-		$reply = $wallMessage->addNewReply( $body, $this->wg->User );
 
-		if ( $reply === false ) {
+		try {
+			$builder =
+				( new WallMessageBuilder() )
+					->setMessageAuthor( $this->getContext()->getUser() )
+					->setMessageText( $body )
+					->setParentMessage( $wallMessage )
+					->setParentPageTitle( $wallMessage->getArticleTitle() );
+
+			$reply = $builder->build();
```

Comment:
I added it mainly for readability to separate it from the setup part where we populate it with data, but yeah, that is the sole purpose